apps/spruce/src/pages/waterfall/WaterfallGrid.tsx

```diff
@@ -114,6 +115,24 @@ export const WaterfallGrid: React.FC<WaterfallGridProps> = ({
     },
   );
 
+  // If we've gotten to the first page, we should erase any minOrder / maxOrder query params.
+  useEffect(() => {
+    const { flattenedVersions, pagination } = data.waterfall;
+    if (minOrder > 0 && flattenedVersions.length > 0) {
+      const activeVersions = flattenedVersions.filter((v) => v.activated);
+      const mostRecentCommitOnPage =
+        flattenedVersions[0].order === pagination.mostRecentVersionOrder;
```

Comment:
Difficult, but did you manage to test this IRL with a new version coming in instead of mocking this scenario using the datepicker? I wonder if that would lead to `mostRecentVersionOrder` being out of date, and thus "hiding" the fact that a new version is present.

If this is in fact the case, one idea is that we could set `nextFetchPolicy` to `cache-and-network` to pull down updated data while still leveraging the cache.

Reply:
Or, maybe polling is likely to update `mostRecentVersionOrder` frequently enough that this is a moot issue and we should avoid complicating things. 🤷‍♀️

Reply:
yes I think the `mostRecentVersionOrder` is updated, which makes this work; I just tested this using the following process

* Go to Cloud project and go 1 page forward.
* Wait and observe GitHub repo to see when a new commit has come in.
* Paginate backwards 1 page. Although this used to be the 0th page, a new commit came in, so the previous page button should not be disabled.
* Click the previous page button. The page is populated with all 5 expected commits.

I'll post a video on the ticket since I don't want to show the commits publicly